src/main/scala/com/persona/service/chat/ChatStorageActor.scala

```diff
@@ -3,22 +3,31 @@ package com.persona.service.chat
 import java.util.UUID
 
 import akka.actor.Actor
-import com.persona.service.chat.dao.MsgHistoryDAO
+import com.persona.service.chat.dao.ChatDAO
+
+import scala.concurrent.ExecutionContext.Implicits._
 
 sealed trait DBEvent
 case class FetchHistory(offerId: UUID, userid: String) extends DBEvent
 case class PersistMsg(offerId: UUID, userid: String, msg: ChatMessage) extends DBEvent
 
-
-class ChatStorageActor extends Actor {
+class ChatStorageActor(chatDAO: ChatDAO) extends Actor {
 
   override def receive = {
     case evt: FetchHistory =>
-      val history = MsgHistoryDAO.fetchMsgHistory(evt.offerId, evt.userid)
-      history.foreach(row => sender() ! HistoryMessage(evt.userid, row.message))
+      val roomActor = sender()
+      chatDAO.fetchMsgHistory(evt.offerId, evt.userid)
+        .onSuccess {
+          case msgs => {
```

Comment:
Not sure why this is a pattern match.  This pattern matches everything.  It would be better to drop the "case" statement